src/main/types/SIOP.types.ts

```diff
@@ -10,38 +10,48 @@ export const expirationTime = 10 * 60;
 
 // https://openid.net/specs/openid-connect-self-issued-v2-1_0.html#section-8
 export interface AuthenticationRequestOpts {
+  authorizationEndpoint?: string;
   redirectUri: string; // The redirect URI
   requestBy: ObjectBy; // Whether the request is returned by value in the URI or retrieved by reference at the provided URL
   signatureType: InternalSignature | ExternalSignature | SuppliedSignature | NoSignature; // Whether no signature is being used, internal (access to private key), or external (hosted using authentication), or supplied (callback supplied)
   responseMode?: ResponseMode; // How the URI should be returned. This is not being used by the library itself, allows an implementor to make a decision
   responseContext?: ResponseContext; // Defines the context of these opts. Either RP side or OP side
+  responseTypesSupported?: ResponseType[];
   claims?: ClaimOpts; // The claims
   registration: RequestRegistrationOpts; // Registration metadata options
   nonce?: string; // An optional nonce, will be generated if not provided
   state?: string; // An optional state, will be generated if not provided
-
+  scopesSupported?: Scope[];
+  subjectTypesSupported?: SubjectType[];
+  requestObjectSigningAlgValuesSupported?: SigningAlgo[];
   // slint-disable-next-line @typescript-eslint/no-explicit-any
   // [x: string]: any;
 }
 
-// https://openid.net/specs/openid-connect-implicit-1_0.html#AuthenticationRequest
+// https://openid.bitbucket.io/connect/openid-connect-self-issued-v2-1_0.html#section-10
 export interface AuthenticationRequestPayload extends JWTPayload, RequestRegistrationPayload {
   scope: string;
   response_type: ResponseType;
   client_id: string; // did of RP
   redirect_uri: string;
   id_token_hint?: string; // TODO:  idtokenhint parameter value, as specified in Section 3.1.2. If the ID Token is encrypted to the Self-Issued OP, the sub (subject) of the signed ID Token MUST be sent as the kid (Key ID) of the JWE.
   // iss: string;
-  response_mode: ResponseMode;
-  response_context: ResponseContext;
-
+  //response_mode: ResponseMode;
```

Comment:
Hmzz, I believe Response_mode wat a lib implementation detail, not sure if it can be removed right now

Reply:
added back